packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -231,9 +231,11 @@ export const getCartDetails = (payload = {}) => {
 
             // for each item in the cart, look up its image in the cache
             // and merge it into the item object
+            // then assign its options from the totals subResource
             if (imageCache && Array.isArray(items) && items.length) {
-                items.forEach(item => {
+                items.forEach((item, i) => {
                     item.image = item.image || imageCache[item.sku] || {};
+                    item.options = totals.items ? totals.items[i].options : [];
```

Comment:
Are we guaranteed that the order of entries in `details.items` is the same as the order in `totals.items`?

If not, we might be mixing up `options` here.

Reply:
```suggestion
                    item.options = totals.items.find(t => t.item_id === item.item_id).options;
```
Interesting point - although it does seem to always be in the same order, I suppose I don't know for sure that it will always be. How about matching the item Id instead?